### ISSUE TITLE
Cannot save sources on Violation Location fields

### ISSUE THREAD

**Issue by jeancochrane** [security-force-monitor/sfm-cms#605]
Currently, all of the Violation Location fields that are exposed in the `edit-violation-locations` form (`Location`, `LocationDescription`, `AdminLevel1`, and `AdminLevel2`) have `source_required == False`. Because of the way that forms are implemented in the app, this means that the sources for these fields won't be saved, even though the UI allows you to add sources (and in fact, it will throw an error if you don't). Relevant code block here:

https://github.com/security-force-monitor/sfm-cms/blob/7dc9fa23c9da8dcdbadd4b4fe1a2de2efabbc4e4/sfm_pc/forms.py#L354-L357

Two ways to fix this:

1. Remove the option to add sources for these fields in the UI
2. Set `source_required == True` for these fields, which may then require updating the importer

What do you think @tlongers @tonysecurityforcemonitor? In general, I don't quite understand why Violation Basics fields don't require sources, so 2) seems like the better fix to me.

**Comment by tonysecurityforcemonitor** [comment]
In our model Incidents/Violations all come from a single source, so those fields don't really have a related "source field" as compared to Org:Name, Org:Membership, etc.

So I would think option 1 would be the way to go... as long as keeping the source attached to the description still carries through

**Comment by tlongers** [comment]
Thanks for the work on this @jeancochrane.  As @tonysecurityforcemonitor outlined incidents  are conceptually different because they are simply data drawn from a single source, reflecting how a specific single group (like Amnesty, or the UNHCHR) has reported something; our other data are composites from many sources. Unusually, it seems that what has caught us out here is that incidents are simpler to enter than everything else, so we need to hide functionality.

The reasons we approach incidents this way are outlined in the [Research Handbook](https://help.securityforcemonitor.org/v/en/data-model/events), but reading that gives no indication of how that  might impact the engineering side. 

I did attempt to show in the Wireframes how this would affect the entry of incident data (page 7, [here](https://securityforcemonitor.org/wp-content/uploads/2018/09/data-entry-ui.pdf)):

![image](https://user-images.githubusercontent.com/2115933/64544831-980dd100-d31f-11e9-8130-ebe1638befce.png)

The idea would be that the picker would just be stuck on the description field, enabling the selection of one source only. Behind the scenes the system would just duplicate this for all the other fields, and the user not have to do anything.

This is in line with your  option `1` above - it's the best approach for now. Perhaps "grey out" the source picker for everything except the `description` field.

Down the line, we would just drop the multi-source, multi-tabbed approach entirely for incidents and have a single form, atop which we select one source and then just add in the data and locations and remove any sort of ambiguity about what the user needs to do. That, however, may not be possible in the current round.

**Comment by jeancochrane** [comment]
Thanks for the clear explanation @tlongers! My bad for moving forward with a solution without clearing it with you two. Seems like I must have missed @tonysecurityforcemonitor's comment above, too, which also makes it clear that option 1 is the preferred solution above option 2.

I'll take some time to reorient these views so that `Description` is the only field requiring sources. Depending on how things look it may actually be simpler to refactor the forms to remove the tabbed view. Whatever happens, I'll do my best to share a draft of it with you two quickly so that we can validate the approach.

**Comment by jeancochrane** [comment]
Alright, I pushed up some tweaks in https://github.com/security-force-monitor/sfm-cms/commit/3317f4a7a78b0974a7043b9c974b6dc1f1669479 that should return us back to the state of only needing sources for `violation:description`. I think there are some improvements we could make to the form/tab views to make the sourcing clearer, but in the spirit of focusing only on basic functionality I decided to revert to the previous `Basics` view and always show the source picker tied to the `Description` field, and then hide the source picker altogether on the `Locations` tab. Let me know how it looks!
